src/warning.js

```diff
@@ -0,0 +1,29 @@
+QUnit.warning = (function() {
+    var deprecated = {
+        asyncTest: "DEPRECATED: QUnit.asyncTest() will " +
+            "be removed in QUnit 2.0.",
+        init: "DEPRECATED: QUnit.init() will throw " +
+            "an error in 2.0, and will be removed in 2.1",
+        module: "DEPRECATED: Global 'module()' will " +
+            "be removed in QUnit 2.0, use 'QUnit.module() instead" +
+            "the assertion context.",
+        push: "DEPRECATED: QUnit.push() is deprecated and " +
+            "it's recommended to use it through its direct reference " +
+            "in the assertion context.",
+        stop: "DEPRECATED: QUnit.stop() will be removed in QUnit 2.0.",
+        test: "DEPRECATED: Global 'test()' will be removed in " +
+            "QUnit 2.0, use 'QUnit.test() instead"
+    };
+
+    function log(warningMessage) {
+        if ( global.console && global.console.warn ) {
+            if (warningMessage) {
```

Comment:
you may blend this warningMessage to the if statement above. It cost less to check for it before at first too.